controllers/template_sync.go

```diff
@@ -335,3 +338,71 @@ func overrideRemediationAction(instance *policiesv1.Policy, tObjectUnstructured
 		}
 	}
 }
+
+// emitTemplateError performs actions that ensure correct reporting of template errors in the
+// policy framework. If the policy's status already reflects the current error, then no actions
+// are taken.
+func (r *PolicyReconciler) emitTemplateError(pol *policiesv1.Policy, tIndex int, tName, errMsg string) {
+	// check if the error is already present in the policy status - if so, return early
+	if strings.Contains(getLatestStatusMessage(pol, tIndex), errMsg) {
+		return
+	}
+
+	// emit the non-compliance event
+	policyComplianceReason := fmt.Sprintf(policyFmtStr, pol.GetNamespace(), tName)
+	r.Recorder.Event(pol, "Warning", policyComplianceReason, "NonCompliant; template-error; "+errMsg)
+
+	// emit an informational event
+	r.Recorder.Event(pol, "Warning", "PolicyTemplateSync", errMsg)
+}
+
+// handleSyncSuccess performs common actions that should be run whenever a template is in sync,
+// whether there were changes or not. If no changes occurred, an empty message should be passed in.
+// If the given policy template was in a template-error state (determined by checking the status),
+// then the template object's `status.compliant` field (complianceState) will be reset. When this
+// occurs, the relevant policy controller must re-populate it, and emit a new compliance event for
+// the framework to observe.
+func (r *PolicyReconciler) handleSyncSuccess(
+	ctx context.Context,
+	pol *policiesv1.Policy,
+	tIndex int,
+	tName, msg string,
```

Comment:
This confused me when looking to where `msg` was defined.  :smile: 
```suggestion
	tName string,
	msg string,
```